targets/TARGET_STM/sleep.c

```diff
@@ -200,6 +201,10 @@ void hal_deepsleep(void)
     HAL_PWR_EnterSTOPMode(PWR_LOWPOWERREGULATOR_ON, PWR_STOPENTRY_WFI);
 #endif /* TARGET_STM32L4 */
 
+    /* Prevent HAL_GetTick() from using ticker_read_us() to read the
```

Comment:
What about moving  restore_timer_ctx(); at this stage so that HAL_GetTick can be called in the next functions ? 
Have you tried this before

Reply:
The following functions:
1. `ForceClockOutofDeepSleep()`,
1. `SetSysClock()`,
1. `restore_timer_ctx()`,

have to be called in that order. If the timer context was restored earlier, it would have been reset anyway.

Reply:
Ok good point indeed. will approve as it is